Rework Python API

rgieseke opened the pull request:
Proposal for #3:

Dictionary with None values dropped:
```py
globalwarmingpotentials = {
    "SARGWP100": {
        "CH4": 21.0,
        "N2O": 310.0,
# ...
```

For getting fields one can use `globalwarmingpotentials.keys()` or `globalwarmingpotentials["SARGWP100"].keys()`

If Pandas is installed a DataFrame can be loaded with
```py
df = globalwarmingpotentials.as_data_frame()
```

```
In [2]: globalwarmingpotentials.as_data_frame()
Out[2]: 
           SARGWP100  AR4GWP100  AR5GWP100  AR5CCFGWP100
Species                                                 
CH4             21.0       25.0       28.0            34
N2O            310.0      298.0      265.0           298
CFC11         3800.0     4750.0     4660.0          5352
```

rgieseke committed:
Rework Python API

danielhuppmann commented on src/globalwarmingpotentials/__init__.py:
```diff
@@ -9,370 +9,306 @@
-    "CH2Cl2": 11,
-    "CH3Cl": 15,
-    "Halon1201": 454,
+globalwarmingpotentials = {
```
This would mean that the API is
```python
globalwarmingpotentials.globalwarmingpotentials
```
which is not quite elegant. Hence the suggestion for a shorter name...

```suggestion
data = {
```

danielhuppmann replied:
This way, the nested dict could be retrieved as
```python
import globalwarmingpotentials as gwp
gwp.data
```

rgieseke replied:
Thanks for the feedback! :-)

Still more to type, but how about 
```py
from globalwarmingpotentials import globalwarmingpotentials as gwp

gwp["AR5GWP100"]
```

danielhuppmann replied:
Well, having a package and an attribute of the same name could be confusing, but I'll not veto the PR over it...

rgieseke replied:
I think it's the case for some (older?) Python packages, but i agree that it could be confusing :-)

danielhuppmann commented on src/globalwarmingpotentials/__init__.py:
```diff
@@ -9,370 +9,306 @@
-    "CH3Cl": 13.0,
-    "Halon1201": None,
-}
+def as_data_frame():
```
Echoing the equivalent `pandas.Series.to_frame()` method, see [here](https://pandas.pydata.org/docs/reference/api/pandas.Series.to_frame.html)

```suggestion
def as_frame():
```

rgieseke replied:
Nice!

danielhuppmann reviewed (commented):
Looks good, two suggestions inline...

rgieseke committed:
Improve function names

- shorten function name in line with Pandas convention
- avoid same name for module and dict
- add doc string

Co-authored-by: Daniel Huppmann <dh@dergelbesalon.at>

danielhuppmann reviewed (approved):
Looks good, many thanks @rgieseke!

rgieseke committed:
Add example usage to Readme
